fecgov/openFEC
# Refactor ofec_filings_amendments_all_mv

## Summary (required)

- Resolves #[_4276_](https://github.com/fecgov/openFEC/issues/4276)

- delete ofec_processing_non_financia_amendment_chain_vw
- delete ofec_amendments_mv and ofec_amendments_vw
- refactor ofec_filings_amendments_all_mv

## How to test the changes locally
1)  download the branch to local
2) run pytest
3) after running flyway on local, check cfdm_test and these three database objects should be deleted.
      `ofec_processing_non_financia_amendment_chain_vw`
      `ofec_amendments_vw`
      `ofec_amendments_mv`
4) set SQLA_CONN=postgresql://xxx:xxx@/cfdm_test, and run filing endpoint.  ( need to drop/create cfdm_test, invoke sample_db)

( This endpoint uses ofec_filings_all_mv, which joins to ofec_filings_amendments_all_mv)

http://127.0.0.1:5000/v1/filings/?sort=-receipt_date&sort_null_only=false&sort_nulls_last=false&per_page=20&sort_hide_null=false&page=1&form_type=F1
